phpBB/phpbb/language/loader/extension_language_file_loader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\language\loader;
+
+/**
+ * Language loader for phpBB's core language files
+ */
+class extension_language_file_loader extends language_file_loader_base
+{
+	/**
+	 * @var \phpbb\extension\manager	Extension manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * @var string
+	 */
+	protected $extension_name;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\extension\manager	$manager	Extension manager
+	 */
+	public function __construct(\phpbb\extension\manager $manager)
+	{
+		$this->extension_manager = $manager;
+	}
+
+	/**
+	 * Extension name setter
+	 *
+	 * @param string	$extension_name	Name of the extension to load from
+	 */
+	public function set_extension_name($extension_name)
+	{
+		$this->extension_name = $extension_name;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function load($component, $locale, &$lang, &$help)
+	{
+		if (!is_array($locale))
+		{
+			$this->locale = array($locale);
+		}
+		else
+		{
+			$this->locale = $locale;
```

Comment:
we pass the locale via the parameters, so lets remain stateless here and pass them to the parent call aswell